library/trans.c

```diff
@@ -282,52 +282,36 @@ void dnet_trans_destroy(struct dnet_trans *t)
 		localtime_r((time_t *)&t->start.tv_sec, &tm);
 		strftime(str, sizeof(str), "%F %R:%S", &tm);
 
-		if ((t->command == DNET_CMD_READ) && (t->alloc_size >= sizeof(struct dnet_cmd) + sizeof(struct dnet_io_attr))) {
+		if ((t->command == DNET_CMD_READ || t->command == DNET_CMD_WRITE) &&
+		    (t->alloc_size >= sizeof(struct dnet_cmd) + sizeof(struct dnet_io_attr))) {
```

Comment:
`t->alloc_size` and `t` itself is not about the reply but about the request of transaction which in turn has `dnet_io_attr` for write command.

For read command there is [the code](https://github.com/reverbrain/elliptics/blob/master/library/net.c#L690) which copies `dnet_io_attr` from the reply to `t`, so when `t` is destroyed its `dnet_io_attr` will already contain info from the reply.